src/tests/conformance/query/get_by_hash.rs

```diff
@@ -0,0 +1,82 @@
+use tempfile::TempDir;
+
+use crate::{
+    protocol::{
+        codecs::binary::{BinaryMessage, Payload},
+        proto::{
+            tm_get_object_by_hash::ObjectType, TmGetObjectByHash, TmIndexedObject, TmTransactions,
+        },
+    },
+    setup::node::build_stateful_builder,
+    tools::{
+        rpc::{get_transaction_info, wait_for_account_data, wait_for_state},
+        synth_node::SyntheticNode,
+    },
+};
+
+#[tokio::test]
+async fn should_get_transaction_by_hash() {
+    // ZG-CONFORMANCE-008
+    // Create stateful node.
+    let target = TempDir::new().expect("unable to create TempDir");
+    let mut node = build_stateful_builder(target.path().to_path_buf())
+        .expect("unable to get stateful builder")
+        .start(false)
+        .await
+        .expect("unable to start stateful node");
+
+    // Wait for correct state and account data.
+    wait_for_state("proposing".into()).await;
+    let account_data = wait_for_account_data("rNGknFCRBZguXcPqC63k6xTZnonSe6ZuWt") // TODO make constant
+        .await
+        .expect("unable to get account data");
+
+    // Get transaction info by rpc to put in cache.
+    let tx = account_data.result.account_data.previous_transaction;
+    let _ = get_transaction_info(tx.clone())
+        .await
+        .expect("unable to get transaction info");
+
+    // Query transaction via peer protocol.
+    let tx_hash = from_hex_to_bytes(&tx).expect("unable to change transaction hash into bytes");
+    let payload = Payload::TmGetObjectByHash(TmGetObjectByHash {
+        r#type: ObjectType::OtTransactions as i32,
+        query: true,
+        seq: Some(1),
+        ledger_hash: None,
+        fat: None,
+        objects: vec![TmIndexedObject {
+            hash: Some(tx_hash),
+            node_id: None,
+            index: None,
+            data: None,
+            ledger_seq: None,
+        }],
+    });
+    let mut sn = SyntheticNode::new(&Default::default()).await;
```

Comment:
Nit: to be consistent across the codebase.
```suggestion
    let mut synth_node = SyntheticNode::new(&Default::default()).await;
```